### PR TITLE
ci: twister: split publishing results out of main workflow

### PR DESCRIPTION
**Pull request by nashif**
Move publishing data to ES to a workflow_run step. This way we can
change the main workflow to use pull_request instead of
pull_request_target.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]
cool

**Review by stephanosio — DISMISSED** [review]
Will not block this since it is not actually changing the twister workflow to `pull_request` -- when we do that in the future, we need to make sure that the triggering workflow run is validated, similar to how it is done in `doc-publish-pr`.

**Comment by nashif** [comment]
> Will not block this since it is not actually changing the twister workflow to `pull_request` -- when we do that in the future, we need to make sure that the triggering workflow run is validated, similar to how it is done in `doc-publish-pr`.

that split workflow_run  should not run on pull_requests though.

**Review by bjarki-andreasen — APPROVED** [review]

File: .github/workflows/twister-prep.yaml [hunk 1]
```diff
@@ -0,0 +1,150 @@
+name: Prep
```
Comment (stephanosio) [inline .github/workflows/twister-prep.yaml:1]:
```suggestion
name: Twister Prep
```

to make it more clear what it is in the workflow list ...

**Review by stephanosio — APPROVED** [review]

**Pull request merged**